src/loader/load_sonar_dataset.py

```diff
@@ -101,8 +104,13 @@ def timestamp_to_microseconds(timestamp: str):
     def label_timestamp_to_microseconds(label_obj: dict):
         label_obj["timeStarted"] = timestamp_to_microseconds(label_obj["timeStarted"])
         return label_obj
+    
+    def str_label_to_activity_idx(label_obj: dict):
+        label_obj["label"] = settings.DATA_CONFIG.raw_label_to_activity_idx(label_obj["label"])
+        return label_obj
```

Comment:
Kirill and I had a discussion about this. I think in the end, we agreed, that it makes no sense to replace the activity strings with IDs at this point, as it makes e.g. plotting and rewriting labels way harder.

Reply:
Imho a function like this applied before training would make more sense:
```python
def fn(recordings: list[Recording]) -> tuple[list[Recording], dict[str, int]]:
    # Take the set of all activities
    activities = set(itertools.chain(*[recording.activities for recording in recordings]))
    # Generate an id for every activity
    activity_str_to_id_dict = {v: idx for idx, v in enumerate(activities)}
    # Map all recording activities to these ids
    for recording in recordings:
        recording.activities.replace(activity_str_to_id_dict, inplace=True)

    return recordings, activity_str_to_id_dict
```

Advantages:
- The range of IDs depends on the actual number of activities used (e.g. we take only 4 activities out of our set of 21 -> The activities are labeled in `[0, 1, 2, 3]`. With a static mapping it could be `[0, 5, 13, 20]` which causes errors with `to_categorical`
- The string labels can be kept, if e.g. only necessary for plotting

Disadvantage:
- One line to do the mapping
- Dynamic activity dict has to be passed to functions that need it